### PR TITLE
Ts migration part4

### PR DESCRIPTION
**Pull request by hiredgun**
Migrates
- main files of subscriptions and streaming 
- connection

### CODE REVIEW

File: src/openapi/streaming/subscription.ts [hunk 1]
```diff
@@ -0,0 +1,1131 @@
...
+    UPDATE_TYPE_SNAPSHOT = 1 as const;
+    UPDATE_TYPE_DELTA = 2 as const;
+    // FIXME consider making those constants static - BREAKING CHANGE
+    STATE_SUBSCRIBE_REQUESTED = 1 as const;
+    STATE_SUBSCRIBED = 2 as const;
+    STATE_UNSUBSCRIBE_REQUESTED = 4 as const;
+    STATE_UNSUBSCRIBED = 8 as const;
+    STATE_PATCH_REQUESTED = 16 as const;
+    // FIXME this const was originally undefined - verify why
+    STATE_READY_FOR_UNSUBSCRIBE_BY_TAG = 32 as const;
```
Comment (hiredgun) [inline src/openapi/streaming/subscription.ts:95]: Please pay attention to this change and try to verify that I didn't make sth worse by introducing the missing flag.

Comment (hiredgun) [reply src/openapi/streaming/subscription.ts:95]: By adding the missing constant I think the behaviour is now change in [this line](https://github.com/hiredgun/openapi-clientlib-js/pull/28/files#diff-4efe8e9679cdce129f826267dd40340a26926acd7730c56c50d433cf86cbeee9R327) but I've found this PR and it seems like STATE_READY_FOR_UNSUBSCRIBE_BY_TAG was overlooked.

File: src/openapi/streaming/streaming.spec.ts [hunk 2]
```diff
@@ -81,24 +102,34 @@ describe('openapi Streaming', () => {
     afterEach(() => uninstallClock());
 
     function mockSubscription() {
-        function changeState(state) {
-            this.stateChangedCallback.forEach((callback) => callback(state));
-        }
+        type Callback = (state: SubscriptionState) => void;
+
+        // function changeState(state) {
```
Comment (Sahilgoyal190) [inline src/openapi/streaming/streaming.spec.ts:107]: Why this is being commented?

File: src/openapi/streaming/streaming.spec.ts [hunk 3]
```diff
@@ -944,25 +978,28 @@ describe('openapi Streaming', () => {
...
             streaming.subscriptions.push(subscription2);
 
+            // @ts-expect-error
             streaming.disposeSubscription(subscription);
 
             expect(subscription.onUnsubscribe.mock.calls.length).toEqual(1);
             expect(subscription.dispose.mock.calls.length).toEqual(1);
             expect(streaming.subscriptions.length).toEqual(1);
-
+            // @ts-expect-error
```
Comment (Sahilgoyal190) [inline src/openapi/streaming/streaming.spec.ts:994]: can we add comments why we are expecting error in all these places

File: src/openapi/streaming/streaming.spec.ts (hunk 2 above)
Comment (hiredgun) [reply src/openapi/streaming/streaming.spec.ts:107]: it was used in lin 132 but the changeState property from line 132 was not used anywhere so I temporarily commented it out but forgot to remove :) Will remove it

File: src/openapi/streaming/subscription.ts (hunk 1 above)
Comment (Sahilgoyal190) [reply src/openapi/streaming/subscription.ts:95]: May be you can check with owner of PR or Luke

**Review by Sahilgoyal190 — CHANGES_REQUESTED** [review]

File: src/openapi/streaming/subscription.ts (hunk 1 above)
Comment (hiredgun) [reply src/openapi/streaming/subscription.ts:95]: Talked to Luke, will leave my fix and will try to cover this case with unit tests in a separate PR

**Review by Sahilgoyal190 — APPROVED** [review]

**Pull request merged**